docs/reference/pip_install.rst

```diff
@@ -16,14 +16,17 @@ Description
 .. pip-command-description:: install
 
 
-.. _`Requirements File Format`:
-
 Installation Order
 ++++++++++++++++++
 
-Pip installs dependencies before the things that depend on them. In the event
-of a dependency cycle, the first encountered member of the cycle is installed
-last.
+As of v6.1.0, pip installs dependencies before their dependents, i.e. in
+"topological order".  This is the only commitment pip currently makes related
+to order.  While it may be coincidentally true that pip will install things in
+the order of the install arguments or in the order of the items in a
+requirements file, this is not a promise.
+
+In the event of a dependency cycle (aka "circular dependency"), the first
+encountered member of the cycle is installed last.
```

Comment:
We might want to note that this is the current implementation and not a long term promise though.